fastcov.py

```diff
@@ -184,55 +184,68 @@ def dumpToLcovInfo(fastcov_json, output):
             f.write("LH:{}\n".format((len(data["lines"]) - line_miss)))   #Lines Hit
             f.write("end_of_record\n")
 
-def exclMarkerWorker(fastcov_sources, chunk, exclude_branches_sw, include_branches_sw):
+def getSourceLines(source, fallback_encodings=[]):
+    """Return a list of lines from the provided source, trying to decode with fallback encodings if the default fails"""
+    default_encoding = sys.getdefaultencoding()
+    for encoding in [default_encoding] + fallback_encodings:
+        try:
+            with open(source, encoding=encoding) as f:
+                return f.readlines()
+        except UnicodeDecodeError:
+            pass
+
+    log("Warning: could not decode '{}' with {} or fallback encodings ({}); ignoring errors".format(source, default_encoding, ",".join(fallback_encodings)))
+    with open(source, errors="ignore") as f:
+        return f.readlines()
+
+def exclMarkerWorker(fastcov_sources, chunk, exclude_branches_sw, include_branches_sw, fallback_encodings):
     for source in chunk:
         # If there are no covered lines, skip
         if not fastcov_sources[source]["lines"]:
             continue
 
         start_line = 0
         end_line = 0
-        with open(source) as f:
-            for i, line in enumerate(f, 1): #Start enumeration at line 1
-                if str(i) in fastcov_sources[source]["branches"]:
-                    if all(not line.lstrip().startswith(e) for e in include_branches_sw): # Include branches starting with...
-                        del fastcov_sources[source]["branches"][str(i)]
+        for i, line in enumerate(getSourceLines(source, fallback_encodings), 1): #Start enumeration at line 1
+            if str(i) in fastcov_sources[source]["branches"]:
+                if include_branches_sw and all(not line.lstrip().startswith(e) for e in include_branches_sw): # Include branches starting with...
+                    del fastcov_sources[source]["branches"][str(i)]
+
+                if exclude_branches_sw and any(line.lstrip().startswith(e) for e in exclude_branches_sw): # Exclude branches starting with...
```

Comment:
`exclude_branches_sw and` not needed here, but more explicit.

Reply:
Yes, I made them both explicit since I was burned previously by the implicit behavior of `all([])` vs `any([])` :P